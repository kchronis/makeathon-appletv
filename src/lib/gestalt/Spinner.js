/**
 * WIP - Gestalt like Spinner
 * https://github.com/pinterest/gestalt/tree/master/src/Spinner
 * @flow
 */

import * as React from 'react';
import { View, requireNativeComponent } from 'react-native';
import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import Text from './Text';

import PIColor from '../styles/picolor';

const ReactNativeIndeterminateActivityIndicatorView = requireNativeComponent(
  'PIReactNativeIndeterminateActivityIndicatorView',
  null
);

const SIZE = 64;

type Props = {|
  accessibilityLabel: string,
  show: boolean,

  style?: StyleObj, // React Native only: Should go away eventually
|};

export default function Spinner({ accessibilityLabel, show, style }: Props) {
  return (
    <ReactNativeIndeterminateActivityIndicatorView
      accessibilityLabel={accessibilityLabel}
      show={show}
      style={[{ width: SIZE, height: SIZE }, style]}
    />
  );
}

// Playground support

// eslint-disable-next-line import/first
import { type PlaygroundDefineType } from '../../screens/Playground/Playground';

module.exports.__cards__ = (define: PlaygroundDefineType) => {
  define('Spinner Default', () => (
    <View
      style={{
        backgroundColor: PIColor.primaryBackgroundColor,
        flex: 1,
        justifyContent: 'center',
        padding: 10,
      }}
    >
      <Text style={{ paddingBottom: 5 }}>Spinner:</Text>
      <Spinner
        accessibilityLabel={'Label Haha'}
        show
        style={{ alignSelf: 'center' }}
      />
    </View>
  ));
};
