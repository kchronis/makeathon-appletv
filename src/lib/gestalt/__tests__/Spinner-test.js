/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../Spinner';

test('Spinner visible', () => {
  const component = renderer.create(
    <Spinner accessibilityLabel={'Label Haha'} show />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
