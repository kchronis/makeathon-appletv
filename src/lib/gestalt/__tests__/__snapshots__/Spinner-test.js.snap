// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner visible 1`] = `
<PIReactNativeIndeterminateActivityIndicatorView
  accessibilityLabel="Label Haha"
  show={true}
  style={
    Array [
      Object {
        "height": 64,
        "width": 64,
      },
      undefined,
    ]
  }
/>
`;
